lib/rubygems/package/tar_reader/entry.rb

```diff
@@ -83,7 +83,7 @@ def getc
     return nil if @read >= @header.size
 
     ret = @io.getc
-    @read += 1 if ret
+    @read += char_bytesize(ret) if ret
```

Comment:
Rather than add this, I think it would be better to change the above `@io.getc` to `@io.getbyte`.  `@io.getbyte` will return an integer across all versions of Ruby (1.8 and up), and we don't need to change the rest of the code since the rest of the code assumes integers (presumably).